content/docs/tech-insights/track-dependency-versions/index.md

```diff
@@ -0,0 +1,273 @@
+---
+title: Tracking Dependency versions
+publishedDate: '2024-01-08T15:15:00.0Z'
+description: How to track and visualize the dependency versions of your software with Roadie Tech Insights
+---
+
+Tracking the dependency version is critical for ensuring your software is secure and compliant. It can be difficult to track which of your software is using a supported version, and which are using an End of Life version.
+
+With Roadie’s Tech Insights feature, there’s an easier way. In this tutorial, we’re going to:
+
+1. Automatically scan Components in the Backstage catalog and record which React version they’re using.
+2. Visualise the distribution of React versions being used.
+3. Create a check that can tell us which software is not using the supported version.
+
+As we go through this process, you’ll learn:
+
+1. How to create Data Sources, Checks and Scorecards with Tech Insights.
+2. How to parse files with JSONata in Roadie and extract facts from those files.
+3. How to detect which repositories contain certain files like the package.json with Tech Insights.
+4. How to use the “Satisfies semantic version” operator to do comparisons.
+
+Let’s get started.
+
+<details>
+
+<summary style="font-size: 24px;">React Version</summary>
+
+## Record the React version automatically
+
+In order to track the React versions used in the catalog, we are going to create a Data Source In Roadie Tech Insights. It will be setup to run on a scheduled basis and extract the React version specified in the package.json file.
+
+### Package.json Data Source
+
+1. Visit Tech Insights, select the **Data Sources** tab, and press **ADD DATA SOURCE**.
+
+   ![a button on a web interface](./Add_Data_Source.png)
+
+2. Enter a descriptive **Name** and **Description**.
+
+   ![About section input fields](./react/React_Data_About_Section.png)
+
+3. Set the Data Provider **Type** to `Component Repository File`, **Location** to `package.json`. Select a value for **Select Entity to test data source against** and press **TEST**.
+
+    ![Data Provider section input fields](./react/React_Data_Provider_Section.png)
+
+4. Test results are displayed.
+
+    ![Data Provider test results displayed](./react/React_Data_Test_Results_Section.png)
+
+5. Create multiple Facts:
+    1. Set **Parser** to `JSON with JSONata syntax`.
+    2. Set the first Fact to:
+        | Field Name | Value |
+        | --- | --- |
+        | Fact Name | React version |
+        | JSONata query | $exists(resolutions."@types/react") ? resolutions."@types/react" : $exists(dependencies.react) ? dependencies.react : "" |
```

Comment:
```suggestion
        | JSONata query | $exists(resolutions."react") ? resolutions."react" : $exists(dependencies.react) ? dependencies.react : "" |
```

This is probably better like this so the JS only projects are covered as well

Reply:
Pushing as is, would like to further discuss before making any changes to this.